common/views/components/IIIFViewer/IIIFViewer.js

```diff
@@ -0,0 +1,339 @@
+
+const IIIFViewerPaginatorButtons = styled.div.attrs(props => ({
+  className: classNames({
+    'flex absolute flex--h-center': true,
+  }),
+}))`
+  left: ${props => (props.isThumbs ? 'auto' : '50%')};
+  right: ${props => props.theme.spacingUnit}px;
+  bottom: ${props => (props.isThumbs ? '50%' : props.theme.spacingUnit + 'px')};
+  transform: ${props =>
+    props.isThumbs ? 'translateY(50%)' : 'translateX(-50%)'};
+
+  ${props =>
+    props.isThumbs &&
+    `
+    @media (min-width: ${props.theme.sizes.medium}px) {
+      bottom: ${props.theme.spacingUnit}px;
+      left: 50%;
+      transform: translateX(-50%) translateY(0%);
+
+      .control__inner {
+        transform: rotate(90deg);
+      }
+    }
+  `}
+`;
+
+const IIIFViewerThumbNumber = styled.span.attrs(props => ({
+  className: classNames({
+    'line-height-1': true,
+    'absolute bg-charcoal font-white': true,
+    [font({ s: 'LR3' })]: true,
+  }),
+}))`
+  top: ${props => props.theme.spacingUnit}px;
+  left: 50%;
+  transform: translateX(-50%);
+  padding: 3px 2px 0;
+  z-index: 1;
+`;
+
+const IIIFViewerThumb = styled.div.attrs(props => ({
+  className: classNames({
+    'relative flex flex--v-center': true,
+    [spacing({ s: 1 }, { padding: ['top', 'right', 'bottom', 'left'] })]: true,
+  }),
+}))`
+  height: 100%;
+  width: 20%;
+  margin-right: ${props => props.theme.spacingUnit}px;
+
+  &:last-child {
+    margin: 0;
+  }
+
+  @media (min-width: ${props => props.theme.sizes.medium}px) {
+    height: 25%;
+    width: 100%;
+    margin-right: 0;
+  }
+`;
+
+const IIIFViewerThumbs = styled.div.attrs(props => ({
+  className: classNames({
+    'flex flex--h-center relative bg-smoke': true,
+  }),
+}))`
+  height: 20%;
+  width: 100%;
+  padding: 0 100px 0 0;
+
+  @media (min-width: ${props => props.theme.sizes.medium}px) {
+    height: 100%;
+    flex-direction: column;
+    width: 25%;
+    padding: 0 0 ${props => props.theme.spacingUnit * 10}px;
+  }
+`;
+
+const IIIFViewerMain = styled.div.attrs(props => ({
+  className: classNames({
+    'relative bg-charcoal font-white': true,
+    [spacing({ s: 4 }, { padding: ['top'] })]: true,
+    [spacing({ s: 1 }, { padding: ['right', 'left'] })]: true,
+    [spacing({ s: 10 }, { padding: ['bottom'] })]: true,
+  }),
+}))`
+  height: 80%;
+  width: 100%;
+
+  @media (min-width: ${props => props.theme.sizes.medium}px) {
+    height: 100%;
+    width: 75%;
+  }
+`;
+
+const IIIFViewerXOfY = styled.span.attrs(props => ({
+  className: classNames({
+    'absolute font-white': true,
+    [spacing({ s: 1 }, { margin: ['left', 'right'] })]: true,
+    [font({ s: 'LR3' })]: true,
+  }),
+}))`
+  top: ${props => props.theme.spacingUnit}px;
+  left: 50%;
+  transform: translateX(-50%);
+`;
+
+const IIIFViewerThumbLink = styled.a.attrs(props => ({
+  className: classNames({
+    'block h-center': true,
+  }),
+}))`
+  height: 100%;
+
+  img {
```

Comment:
Is there a reason not to put a component around the image instead?

I'm finding myself confused as what old rules to apply to the new way of doing things.

BEM, avoiding using elements, low/no specificity etc still feel like the reasons for those apply?

Reply:
Same for _Chewing the Code_?